app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -0,0 +1,271 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.viewmodel.compose.viewModel
+import ch.hikemate.app.R
+import ch.hikemate.app.model.route.ListOfHikeRoutesViewModel
+import org.osmdroid.config.Configuration
+import org.osmdroid.events.MapListener
+import org.osmdroid.events.ScrollEvent
+import org.osmdroid.events.ZoomEvent
+import org.osmdroid.util.GeoPoint
+import org.osmdroid.views.MapView
+
+object MapScreen {
+  const val TEST_TAG_MAP = "map"
+  const val TEST_TAG_MENU_BUTTON = "menuButton"
+  const val TEST_TAG_HIKES_LIST = "hikesList"
+  const val TEST_TAG_HIKE_ITEM = "hikeItem"
+}
+
+@Composable
+fun MapScreen(
+    hikingRoutesViewModel: ListOfHikeRoutesViewModel =
+        viewModel(factory = ListOfHikeRoutesViewModel.Factory)
+) {
+  val context = LocalContext.current
+  // Avoid re-creating the MapView on every recomposition
+  val mapView = remember { MapView(context) }
+
+  Configuration.getInstance().apply {
+    // Set user-agent to avoid rejected requests
+    userAgentValue = context.packageName
+
+    // Allow for faster loading of tiles. Default OSMDroid value is 2.
+    tileDownloadThreads = 4
+
+    // Maximum number of tiles that can be downloaded at once. Default is 40.
+    tileDownloadMaxQueueSize = 40
+
+    // Maximum number of bytes that can be used by the tile file system cache. Default is 600MB.
+    tileFileSystemCacheMaxBytes = 600L * 1024L * 1024L
+  }
+
+  mapView.apply {
+    controller.setZoom(15.0)
+    // TODO : Set the center to the user's location
+    controller.setCenter(GeoPoint(46.5, 6.6))
+    // Enable touch-controls such as pinch to zoom
+    setMultiTouchControls(true)
+    // Update hiking routes every time the user moves the map
+    // TODO : Those updates could be quite frequent, have a cooldown to avoid sending a request each
+    // time?
+    addMapListener(
+        object : MapListener {
+          override fun onScroll(event: ScrollEvent?): Boolean {
+            hikingRoutesViewModel.setArea(mapView.boundingBox)
+            return true
+          }
+
+          override fun onZoom(event: ZoomEvent?): Boolean {
+            hikingRoutesViewModel.setArea(mapView.boundingBox)
+            return true
+          }
+        })
+  }
+
+  Box(modifier = Modifier.fillMaxSize()) {
+    AndroidView(
```

Comment:
Nitpick: Perhaps add a comment to explain what AndroidView is

Reply:
Might not be so obvious indeed. Would this comment be a good explanation?

```kotlin
// Jetpack Compose is a relatively recent framework for implementing Android UIs. OSMDroid is
    // an older library that uses Activities, the previous way of doing. The composable AndroidView
    // allows us to use OSMDroid's legacy MapView in a Jetpack Compose layout.
```

Reply:
Yes, that makes it clear.